File: week_1/mario-less/mario.c
#include <stdio.h>
#include <cs50.h>

int validate_input(void);
void pyramide(int n);

int main(void)
{
    // ask user for input and then output pyramide
    int height = validate_input();
    pyramide(height);
}

int validate_input(void)
{
    int n;
    do
    {
        n = get_int("Height: ");
    }
    while (n < 1);
    return n;
}

void pyramide(int n)
{
    for (int i = 0; i < n; i++)
    {
        for (int j = n - 1; j > i; j--)
        {
            printf(" ");

        }
        for (int k = 0; k < (i + 1); k++)
        {
            printf("#");
        }

        printf("\n");
    }
}

File: week_6/sentimental-mario-more/mario.py
import sys

while True:
    try:
        height = int(input("Height: "))
    except ValueError:
        continue

    if height > 0 and height <= 8:
        break


for i in range(0, height):
    for j in range(height-1, i, -1):
        print(" ", end="")

    for j in range(0, i+1):
        print("#", end="")

    print("  ", end="")

    for j in range(0, i+1):
        print("#", end="")
    print()
